pkg/cluster/cluster.go

```diff
@@ -45,25 +45,24 @@ const (
 
 type clusterEvent struct {
 	typ  clusterEventType
-	spec spec.ClusterSpec
+	spec *spec.ClusterSpec
 }
 
 type Config struct {
-	Name          string
-	Namespace     string
 	PVProvisioner string
 	s3config.S3Context
 	KubeCli *unversioned.Client
 }
 
+// TODO: better rename to ClusterManager
 type Cluster struct {
 	logger *logrus.Entry
 
 	Config
 
-	status *Status
+	*spec.EtcdCluster
```

Comment:
why embed this? embed stuff will make them public. we should keep this field private.